Fixed matching URL path to service type in Extensibility.AuthorizeOperationMessageProcessor

SerbanApostol opened the pull request:
Previously, the caller of the constructor had the responsibility to make the path for the service to be lowercase. I modified the constructor to make the path lowercase as it is not obvious to the caller that this is necessary.

Commit:
Fixed a bug when matching URL path to service type
When the dictionary on the constructor of Extensibility.AuthorizeOperationMessageProcessor had its keys not all lowercase, it couldn't find the service type and, consequently, the operation

Commit:
Fixed a bug when matching URL path to service type
When the dictionary on the constructor of Extensibility.AuthorizeOperationMessageProcessor had its keys not all lowercase, it couldn't find the service type and, consequently, the operation

Commit:
Merge branch 'develop' of https://github.com/SerbanApostol/SoapCore into develop

andersjonsson commented:
There is an option to make path insensitive (SoapCoreOption.CaseInsensitivePath). Would it be a good idea to use that to decide if comparison should be case-insensitive?

Dictionary accepts a comparer in the constructor `new Dictionary<string, Type>(StringComparer.OrdinalIgnoreCase)` that makes it easy to vary behavior depending on the option

Commit:
Replaced making each string key lowercase with using StringComparer.OrdinalIgnoreCase

SerbanApostol commented:
Thank you for pointing me to the option of supplying a comparer to Dictionary<,>. I removed the foreach that made each key lowercase and instead used the comparer.

andersjonsson commented:
Sorry for not getting back to you. I haven't had time to work on open source for a while.

I think that you should check the setting SoapCoreOption.CaseInsensitivePath and use that to decide whether to use `StringComparer.Ordinal` or `StringComparer.OrdinalIgnoreCase`

Other than that I think it looks good!

Commit:
Added constructor to Extensibility.AuthorizeOperationMessageProcessor that gets SoapCoreOptions

Added separate constructor to maintain possible backwards compatibility; used options to decide whether to perform case insensitive comparison of paths

Commit:
Configured AuthorizeOperationMessageProcessor in SoapCore.Tests to use CaseInsensitivePath

SerbanApostol commented:
I now notice you mentioned deciding whether to do case insensitive comparison based on the value of SoapCoreOption.CaseInsensitivePath in the first comment as well, sorry about that. I added a constructor that takes an options configuring Action, what I probably should have done the first time around.
I kept the old constructor as well for backwards compatibility, in case anyone else used this in the first place.

andersjonsson commented:
Thanks!
